models/filter/generics.go

```diff
@@ -18,7 +18,7 @@ type SearchModel[TFilter any] struct {
 
 // DefaultsIfNil fills empty but necessary fields with default values
 func (sm *SearchModel[TFilter]) DefaultsIfNil() {
-	if sm.QueryParams == nil {
+	if sm.QueryParams == nil || *sm.QueryParams == (QueryParams{}) {
```

Comment:
WDYT about creating helper method simillar to this?

```go
func (sm *SearchModel[TFilter]) DefaultsIfNil() {
    if sm.QueryParams == nil || sm.QueryParams.isEmpty() {
        sm.QueryParams = DefaultQueryParams()
    }
}

func (qp QueryParams) isEmpty() bool {
    return qp == (QueryParams{})
}
```
is it doing the same?

Reply:
helper method added